src/lib/y2users/linux/useradd_config_reader.rb

```diff
@@ -0,0 +1,88 @@
+# Copyright (c) [2021] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "yast2/execute"
+require "y2users/useradd_config"
+
+module Y2Users
+  module Linux
+    # Reads the useradd configuration from the current system
+    class UseraddConfigReader
+      include Yast::Logger
+      Yast.import "ShadowConfig"
+
+      # Creates a {UseraddConfig} object with the system configuration
+      #
+      # @return [UseraddConfig]
+      def read
+        attrs = read_shadow_config.merge(read_useradd)
+        UseraddConfig.new(attrs)
+      end
+
+    private
+
+      # Command for reading the useradd default values
+      USERADD = "/usr/sbin/useradd".freeze
+      private_constant :USERADD
+
+      # Mapping between useradd keys and {UseraddConfig} attributes
+      USERADD_ATTRS = {
+        group:             "GROUP",
+        home:              "HOME",
+        inactivity_period: "INACTIVE",
+        expiration:        "EXPIRE",
+        shell:             "SHELL",
+        skel:              "SKEL",
+        usrskel:           "USRSKEL",
+        create_mail_spool: "CREATE_MAIL_SPOOL"
+      }.freeze
+      private_constant :USERADD_ATTRS
+
+      # Values from "useradd -D"
+      #
+      # @return [Hash]
+      def read_useradd
+        output = Yast::Execute.on_target!(USERADD, "-D", stdout: :capture)
+        values = output.lines.map(&:strip).map { |line| line.split("=", -1) }.to_h
+        USERADD_ATTRS.keys.map { |attr| [attr, useradd_value(values, attr)] }.to_h
+      rescue Cheetah::ExecutionFailed => e
+        log.warn("Failed to read useradd default values' - #{e.message}")
```

Comment:
no support for issues? I do not like much this behavior as not set default is same as if it failed to read.

Reply:
Well, this is called from the more general Linux reader, which does not report issues right now. So no matter if the `UseraddConfigReader` collects issues, they will be discarded by the caller anyways.

None of the readers use the issues mechanism right now. We can open a separate card to reconsider that (including deciding when to show those errors, etc.). But I think it's out of scope here.

Reply:
JFYI, I plan to add an issue when the username is not specified in AutoYaST.

Reply:
> JFYI, I plan to add an issue when the username is not specified in AutoYaST.

I guess the AutoYaST reader is a special case, since there is indeed a step to show the issues found while reading the profile. But we have no such thing in general for reporting the issues that happened while analyzing the system. Not yet, at least.

If we add it, we need to discuss when and how to present such issues to the user in the current workflow.

Reply:
I just added a comment to https://trello.com/c/08t84wmL/4716-a-better-way-for-reporting-issues-at-the-end-of-installation

Reply:
fine for me